cli/src/cmd/forge/fmt.rs

```diff
@@ -63,7 +63,7 @@ impl FmtArgs {
         let mut paths = self.paths.iter().peekable();
 
         if let Some(path) = paths.peek() {
-            if *path == Path::new("-") || !atty::is(atty::Stream::Stdin) {
+            if *path == Path::new("-") && !atty::is(atty::Stream::Stdin) {
```

Comment:
this way we allow
`cat file.sol | forge fmt -`
`forge fmt - < file.sol`

and it doesn't hang with `forge fmt - `, like it's doing on master

Reply:
@shekhirin does it make sense?

Reply:
@joshieDo yep, agree!